test/end-to-end/company.js

```diff
@@ -1933,4 +1933,195 @@ describe("Company endpoint", () => {
+                bio: "Cool company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            test_random = await Company.create({
+                name: "Random Company",
+                contacts: ["112", "122"],
+                bio: "Random company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            await Account.deleteMany({});
+
+            await Account.create({
+                email: test_user_admin.email,
+                password: await hash(test_user_admin.password),
+                isAdmin: true
+            });
+
+            await Account.create({
+                email: test_user_company.email,
+                password: await hash(test_user_company.password),
+                company: test_company._id
+            });
+
+            await Account.create({
+                email: test_random_user.email,
+                password: await hash(test_random_user.password),
+                company: test_random._id
+            });
+        });
+
+        afterEach(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+        });
+
+        afterAll(async () => {
+            await Company.deleteMany({});
+            await Account.deleteMany({});
+        });
+
+        describe("ID Validation", () => {
+            test("Should fail if id is not a valid ObjectID", async () => {
+                const res = await test_agent
+                    .put("/company/123/edit")
+                    .send()
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+
+            test("Should fail if id is not a valid company", async () => {
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .put(`/company/${id}/edit`)
+                    .send()
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+        });
+
+        describe("Should fail if different user/company", () => {
+            test("Should fail if different user", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_random_user)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "Without permission",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
```

Comment:
In the same way we have for offers, a function to generate a company with default values might be useful. Just a suggestion

Reply:
I did it like this because I don't always want to pass all parameters in the body (maybe the company just wants to edit contacts, for example).

Reply:
This explanation makes sense but in that sense maybe you could write 1 or 2 tests in which you change only a specific attribute, or assert that functionality multiple times in the same test.

Reply:
Ok, I will now only change a few attributes for each test.